crypto/bio/bss_dgram.c

```diff
@@ -577,9 +1015,27 @@ static long dgram_ctrl(BIO *b, int cmd, long num, void *ptr)
             num = ret;
         memcpy(ptr, &data->peer, (ret = num));
         break;
+
     case BIO_CTRL_DGRAM_SET_PEER:
         BIO_ADDR_make(&data->peer, BIO_ADDR_sockaddr((BIO_ADDR *)ptr));
         break;
+
+    case BIO_CTRL_DGRAM_GET_ADDR:
+        ret = BIO_ADDR_sockaddr_size(&data->addr);
+        if (ret == 0) {         /* never set, retrieve it */
+            ret = dgram_get_sockname(b);
+        }
+
+        /* FIXME: if num < ret, we will only return part of an address.
+           That should bee an error, no? */
```

Comment:
Needs resolution

Reply:
Not my code actually.  Happy to fix this in separate patch.